modules/oxide-components/src/main/ts/components/dropdown/internals/positioningUtils.ts

```diff
@@ -0,0 +1,77 @@
+import type { CSSProperties } from 'react';
+
+const getMaxHeight = (side: 'top' | 'bottom', anchorRect: DOMRect, boundaryRect: DOMRect): number => {
+  if (side === 'top') {
+    return anchorRect.top - boundaryRect.top;
+  } else {
+    return boundaryRect.bottom - anchorRect.bottom;
+  }
+};
+
+const getMaxWidth = (side: 'left' | 'right', anchorRect: DOMRect, boundaryRect: DOMRect): number => {
+  if (side === 'left') {
+    return anchorRect.left - boundaryRect.left;
+  } else {
+    return boundaryRect.right - anchorRect.right;
+  }
+};
+
+const getPositionStyles = (
+  anchorRect: DOMRect,
+  anchoredContainerRect: DOMRect,
+  side: 'top' | 'bottom' | 'left' | 'right',
+  align: 'start' | 'center' | 'end',
+  gap: number,
+  boundaryRect: DOMRect = document.documentElement.getBoundingClientRect(),
+): CSSProperties => {
```

Comment:
This function seems to take too many arguments, can we instead change it to take one object argument with 6 properties instead?